convert competition entries to articles

codiebeulaine opened the pull request:
Create action that saves the entry as a draft article on wagtail

codiebeulaine committed:
adding admin, with errors

codiebeulaine committed:
taking out unnecessary line

codiebeulaine commented:
@miltontony keep getting an error with wagtail, review please?

miltontony commented:
What error?

miltontony commented on molo/yourwords/admin.py:
```diff
@@ -0,0 +1,18 @@
+
+
+def ConvertToArticle(model_admin, request, entry):
+    article = ArticlePage.objects.create(null, null)
```
So...not quite how creating pages works in Wagtail..There's a tree structure and all pages need to be added in a tree. Have a look at [these tests](https://github.com/praekelt/molo/blob/develop/molo/core/tests/test_models.py#L52-L71) for an example. According to the SoW, the new article will need to be added to a Section called "True Story" or something.. Let's chat in slack

miltontony replied:
Just another note that creating an article is not enough, you'll need to save it as a draft which looks something like this

```
article.save_revision(
    user=request.user,
    submitted_for_moderation=False,
)
```

codiebeulaine committed:
getting changes

elop' into feature/issue-11-convert-competition-entries-to-articles

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

codiebeulaine committed:
changes

codiebeulaine committed:
test

codiebeulaine committed:
failing test wor

codiebeulaine committed:
failing tests

miltontony committed:
remove db files

miltontony committed:
get test to work

codiebeulaine committed:
took out unessecary line

codiebeulaine commented:
@milton ready for review? Just removed an unessecary (don't know how to spell this) line

miltontony commented on molo/yourwords/tests/fixtures/test.json:
```diff
@@ -0,0 +1,18 @@
+[{
```
I don't think you need this fixture at all anymore. You can just create a user normally.

codiebeulaine committed:
removed fixtures

codiebeulaine commented:
@miltontony removed fixtures, still passing. :+1: ?

miltontony commented:
:+1:  This is the first part - the second part should be a redirect to the "move" page. Should be in another PR
